lang/cs/Org.Apache.REEF.IMRU/OnREEF/Client/REEFIMRUClient.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Collections.Generic;
+using System.Globalization;
+using Org.Apache.REEF.Client.API;
+using Org.Apache.REEF.Driver;
+using Org.Apache.REEF.IMRU.API;
+using Org.Apache.REEF.IMRU.OnREEF.Driver;
+using Org.Apache.REEF.IMRU.OnREEF.Parameters;
+using Org.Apache.REEF.Network.Group.Config;
+using Org.Apache.REEF.Network.Group.Driver;
+using Org.Apache.REEF.Network.Group.Driver.Impl;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Tang.Formats;
+using Org.Apache.REEF.Tang.Implementations.Configuration;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Org.Apache.REEF.Tang.Interface;
+using Org.Apache.REEF.Tang.Util;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.Client
+{
+    /// <summary>
+    /// Implements the IMRU client API on REEF.
+    /// </summary>
+    /// <typeparam name="TMapInput">The type of the side information provided to the Map function</typeparam>
+    /// <typeparam name="TMapOutput">The return type of the Map function</typeparam>
+    /// <typeparam name="TResult">The return type of the computation.</typeparam>
+    public sealed class REEFIMRUClient<TMapInput, TMapOutput, TResult> : IIMRUClient<TMapInput, TMapOutput, TResult>
```

Comment:
Got it and agreed

On Wed, Aug 19, 2015 at 3:22 PM, Markus Weimer <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.IMRU/OnREEF/Client/REEFIMRUClient.cs
> <https://github.com/apache/incubator-reef/pull/385#discussion_r37476135>:
>
> > +using Org.Apache.REEF.Tang.Annotations;
> > +using Org.Apache.REEF.Tang.Formats;
> > +using Org.Apache.REEF.Tang.Implementations.Configuration;
> > +using Org.Apache.REEF.Tang.Implementations.Tang;
> > +using Org.Apache.REEF.Tang.Interface;
> > +using Org.Apache.REEF.Tang.Util;
> > +
> > +namespace Org.Apache.REEF.IMRU.OnREEF.Client
> > +{
> > +    /// <summary>
> > +    /// Implements the IMRU client API on REEF.
> > +    /// </summary>
> > +    /// <typeparam name="TMapInput">The type of the side information provided to the Map function</typeparam>
> > +    /// <typeparam name="TMapOutput">The return type of the Map function</typeparam>
> > +    /// <typeparam name="TResult">The return type of the computation.</typeparam>
> > +    public sealed class REEFIMRUClient<TMapInput, TMapOutput, TResult> : IIMRUClient<TMapInput, TMapOutput, TResult>
>
> Yes, I meant for this class not to be public, but internal. You could
> have a public ConfigurationModule which binds this implementation to the
> interface, as well as other parameters. That is how I did it for the
> HadoopFileSystem.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/385/files#r37476135>.
>